### PR TITLE
Add `fpp-to-dict` wrapper command

### LINKED ISSUE #2701: Implement `fprime-util fpp-to-dict`
| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|  |
---
## Feature Description

Run fpp-to-dict wrapper in fprime-util

### PR DESCRIPTION
**Pull request by thomas-bc**
| | |
|:---|:---|
|**_Related Issue(s)_**| https://github.com/nasa/fprime/issues/2701  |
|**_Has Unit Tests (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Adds in the `fpp-to-dict` command wrapper

## Rationale

Implements https://github.com/nasa/fprime/issues/2701

## Testing/Review Recommendations

```
cd fprime/Ref/Top
fprime-util fpp-to-dict
```

### CODE REVIEW

File: src/fprime/fpp/cli.py [hunk 1]
```diff
@@ -76,14 +106,30 @@ def add_fpp_parsers(
...
-    fpp_to_xml_parser.add_argument("--directory", default=None, help="Output directory")
-    return {"fpp-check": run_fpp_check, "fpp-to-xml": run_fpp_to_xml}, {
+    fpp_to_xml_group = fpp_to_xml_parser.add_argument_group("fpp-to-xml arguments")
+    fpp_to_xml_group.add_argument("-d", "--directory", help="Output directory")
+
+    fpp_to_dict_parser = subparsers.add_parser(
+        "fpp-to-dict", help="Run fpp-to-dict utility", parents=[common], add_help=False
+    )
+    fpp_to_dict_group = fpp_to_dict_parser.add_argument_group("fpp-to-dict arguments")
+    fpp_to_dict_group.add_argument("-d", "--directory", help="Output directory")
```
Comment (LeStarch) [inline src/fprime/fpp/cli.py:124]: Does this work?  I thought adding arguments to different groups resulted in exceptions being raised?  Or am I missing something?

Comment (thomas-bc) [reply src/fprime/fpp/cli.py:124]: I am not sure I understand - `.add_argument_group()` is what we use extensively in the GDS to group arguments in the help text. Grouping arguments in help text is the only thing the documentation says this does (https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument_group)

Maybe you're thinking of [add_mutually_exclusive_group()](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) ?

Comment (LeStarch) [reply src/fprime/fpp/cli.py:124]: On line 118, the `-d`, `--dictionary` flag is added to the fpp_to_xml group.
On line 124, the `-d`, `--dictionary` flag is added to the fpp_to_dict group.

In my dalliance with plugins, I was unable to get the same flag added to different groups.  So either I am missing something (possible) or this code will not work (probably not, as I assume it was tested).

Comment (thomas-bc) [reply src/fprime/fpp/cli.py:124]: The code does work, I believe the difference is that here each group is added to a different parser: 
```
fpp_to_xml_parser = subparsers.add_parser(
        "fpp-to-xml", help="Run fpp-to-xml utility", parents=[common], add_help=False
    )
fpp_to_xml_group = fpp_to_xml_parser.add_argument_group("fpp-to-xml arguments")
 . . .
fpp_to_dict_parser = subparsers.add_parser(
        "fpp-to-dict", help="Run fpp-to-dict utility", parents=[common], add_help=False
    )
```
In the GDS, all argument groups are added to the root `fprime-gds` parser.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**